lib/compiler/src/beam_validator.erl

```diff
@@ -2593,6 +2707,23 @@ mv_args([], _VsA, _VsB, Acc) ->
 merge_fragility(FragileA, FragileB) ->
     cerl_sets:union(FragileA, FragileB).
 
+merge_ms_positions(MsPosA, MsPosB, Vs) ->
+    Keys = if
+               map_size(MsPosA) =< map_size(MsPosB) -> maps:keys(MsPosA);
+               map_size(MsPosA) > map_size(MsPosB) -> maps:keys(MsPosB)
```

Comment:
This is a matter of taste but I think that the code below is clearer:

```suggestion
               true -> maps:keys(MsPosB)
```

Reply:
This is a copy of `merge_regs/4` and I'd like to keep the same style. We can change both later on.